src/ape/api/providers.py

```diff
@@ -271,8 +316,42 @@ def priority_fee(self) -> int:
 
     @property
     def base_fee(self) -> int:
-        block = self._web3.eth.get_block("latest")
-        return block.baseFeePerGas  # type: ignore
+        """
+        Returns the current base fee from the latest block.
+
+        NOTE: If your chain does not support base_fees (EIP-1559),
+        this method will raise a not-implemented error.
+        """
+        block = self.get_block("latest")
+
+        if block.gas_data.base_fee is None:
+            # Non-EIP-1559 chains or we time-travelled pre-London fork.
+            raise NotImplementedError("base_fee is not implemented by this provider.")
+
+        return block.gas_data.base_fee
+
+    def get_block(self, block_id: BlockID) -> BlockAPI:
+        """
+        Returns a block for the given ID.
+
+        Args:
+            block_id: The ID of the block to get. Set as
+              "latest" to get the latest block,
+              "earliest" to get the earliest block,
```

Comment:
you note "earliest" here, but it's not covered in `BlockId`

Also, I can just type 0